packages/cli-core/src/package-manager-controller/package_manager_controller_base.ts

```diff
@@ -96,8 +96,8 @@ export abstract class PackageManagerControllerBase
     const tsConfigTemplate = {
       compilerOptions: {
         target: 'es2022',
-        module: 'es2022',
-        moduleResolution: 'bundler',
+        module: 'preserve',
```

Comment:
Few things here.

1. TS docs mention `esnext` next to `preserve`: `You very likely want "nodenext" for modern Node.js projects and preserve or esnext for code that will be bundled` . Why would we choose `preserve` over `esnext` ? (unless bundling doesn't matter here anymore).
2. `preserve` was added in Typescript 5.4 . If we do this we have to bump Typescript to `^5.4.0`. Including here https://github.com/aws-amplify/amplify-backend/blob/7bf0c64ea43a01e220109d30e98a353864457bf7/packages/create-amplify/src/amplify_project_creator.ts#L21

Reply:
for 1, bundling doesn't matter here aside from Functions. `nodenext` is the one that'll force you to specify `.js` file extensions (though I think there's a new compiler option coming that'll support `.ts` extensions and transform). `preserve` will not transform your module syntax, which should yield the same results when writing with ESM syntax using import/export instead of require/module.exports

for 2, bumped the minimum version